build.py

```diff
@@ -309,28 +315,80 @@ def _download(package: Package, python: Python, dest: str) -> Wheel | None:
             return None
 
 
-def _build(package: Package, python: Python, dest: str) -> Wheel:
+def _likely_binary_exts(sdist: str) -> set[str]:
+    if sdist.endswith(".zip"):
+        with zipfile.ZipFile(sdist) as zipf:
+            names = zipf.namelist()
+    else:
+        with tarfile.open(sdist) as tarf:
+            names = tarf.getnames()
+
+    ret = set()
+    for name in names:
+        if "/test/" in name or "/tests/" in name:
+            continue
+
+        _, ext = os.path.splitext(name)
+        if ext in BINARY_EXTS:
+            ret.add(ext)
+    return ret
+
+
+def _produced_binary(wheel: str) -> bool:
+    with zipfile.ZipFile(wheel) as zipf:
+        for name in zipf.namelist():
+            if name.endswith(".so"):
+                return True
+            # TODO: uwsgi
```

Comment:
I have to handle executable-only dists -- but I'll do that later because they're _complicated_